### DISCUSSION TITLE
Maintainers

### DISCUSSION THREAD

**Discussion by jtr13** [jtr13/cc22mw#4] [General]
Please let me know what your GitHub username is so I can add you as collaborators to the repo

**Comment by jontse** [comment]
Please add me username (jontse).  Thanks!

**Reply by rohansheelvant** [reply]
Please add my username - rohansheelvant. Thanks!

**Comment by philkim72** [comment]
Please add username 'philkim72'

**Reply by philkim72** [reply]
Thanks for adding me.  I see a pull request in the queue.  How do i assign it to myself?

**Reply by philkim72** [reply]
Actually I think I found the button on the right hand side.

**Reply by philkim72** [reply]
I found two issues with pull request #8 and left a comment to fix both issues and requested changes to be made.  I hope I did it properly.

**Reply by jtr13** [reply]
Looks good! I believe the changes have been made so you can review it again.

**Reply by philkim72** [reply]
Thanks for checking.  Btw, was there any further guidance on the bookdown.yml file or should be just use the default method of copying the contents from main and pasting it in to branch copy, then merge.

**Reply by jtr13** [reply]
I just made a short video describing the new merge process. I'll also update the written instructions. 
https://columbiauniversity.zoom.us/rec/play/U0oBpBpNhsT8MT12KW2Tgx3P12gXxWAZYZMUgAmCsn7qX9YhVSJ3tBFIzMq9SsX5RpsDtJjLy8tb_v7T.f9feCnBbS88SsPNG?continueMode=true&_x_zm_rtaid=Bsw20FSZTPGsoztY0ApaKA.1668375087386.cf06b1c3126600adeb86457f357b82a2&_x_zm_rhtaid=959

**Reply by philkim72** [reply]
I'm working on a pull request and the .Rmd file contains install.packages:

```{r}
#install.packages(“quantmod”) 
#install.packages("tidyverse")
#install.packages("tidyquant")
library("quantmod")
library("tidyverse")
library("tidyquant")
```

To confirm, I am supposed to ask them to delete them, and then add them myself to the description file?

**Reply by jtr13** [reply]
No, leave the library statements as is. By adding those libraries to `DESCRIPTION` (if they're not there already) you will be ensuring that those packages are installed. In short, both are needed. Thanks for checking!

**Reply by philkim72** [reply]
Ok thank you.  To confirm, I should add the three packages above to the "Suggests:" section of the description file in the branch, then merge.

I was actually also asking about the "install.packages' statements above, but I also just noticed they were commented out.  If they weren't commented out, I would need to ask them to either delete or comment out, right?

**Reply by jtr13** [reply]
Yes, add to "Suggests:" and ask contributors to delete or comment out `install.packages` statements.

**Reply by philkim72** [reply]
Thanks for confirming.  I noticed that the table of contents on the community contribution page do not include the contribution type (eg cheatsheet, tutorial etc).

**Comment by jtr13** [comment]
The first chapter of every part needs to have a special heading at the top. This explains how to add them: 
https://jtr13.github.io/cc22mw/tutorial-for-pull-request-mergers.html#add-part-names-to-.rmd-for-every-first-article-in-part

**Reply by philkim72** [reply]
I see that the first .Rmd file in each category needs to be amended.  Do we need to fork the repo, create a branch, amend the files, then submit a pull request?

**Reply by jtr13** [reply]
Good question. For things like this you can just edit the file on GitHub and commit directly to the main branch.

**Comment by lauradls** [comment]
Can students include load.packages() in their .rmd file, similarly to how adding library()? Or should we treat it as we treat install.packages() and reject?

**Reply by philkim72** [reply]
Hi Laura, Looks like the following failed from last night:

[Merge pull request #24 from mwreed1/color_deficiency](https://github.com/jtr13/cc22mw/actions/runs/3466993151)

**Reply by lauradls** [reply]
Yes, I'm looking into why per the links on the failed status. Thanks for re-flagging!

**Reply by lauradls** [reply]
Also, when we make changes to the file to fix an error after the book didn't run, is it ok for us to commit directly to the main branch?

**Reply by philkim72** [reply]
I think at this point, you should probably reject the pull request due to the load.packages(), since this may be causing the fails.

**Reply by philkim72** [reply]
And I think it's ok to fix the files directly in main, since it would be too cumbersome to fork the repo etc.  This is consistent with what the prof directed us to do when trying to fix the first files in each category to display the categories.

**Reply by jtr13** [reply]
Yes, it's ok to fix errors directly.

**Reply by jtr13** [reply]
Just to be clear, once a pull request is merged you can't unmerge it if that's what was meant by reject. At the point try to fix the problems and if they can't be fixed then take the `.Rmd` out of the `_bookdown.yml` file so it will stop causing problems. Also create an issue explaining what's going on to keep track of chapters that aren't in the book.

**Comment by rohansheelvant** [comment]
Hello everyone, 

I recently merged a pull request, however the building of bookdown.yaml failed. Attached is the failed message. It maybe linked to the failure of the "color_deficiency.Rmd" file as it contains "load.package" command. Can someone check the failure message and help me with the next steps? 

![failure message](https://user-images.githubusercontent.com/44243145/202032038-be9136f0-fce8-4cf3-b5fb-f693488827ce.png)

**Reply by lauradls** [reply]
Now fixed!

**Reply by philkim72** [reply]
Awesome!

**Comment by philkim72** [comment]
Hello @jtr13 ,

Does this case require special treatment?  The contributor says it needs to be installed from the source (github).

#devtools::install_github("ShubhamKaushal15/bestparcoords")
library("bestparcoords") # Must be installed from source

**Comment by philkim72** [comment]
What is the difference between tutorials and workshops?

**Reply by jtr13** [reply]
In previous semesters students have held workshops either live or virtual. I haven't seen any this semester so you probably won't have any. Remember that you can change around the parts as you see fit; they're not set in stone.

**Comment by jontse** [comment]
Hi All,
I'm surprised to see this being an error in my merge for efficient_r/PR25.  I didn't see any install code, but it's trying to install R.  What am I missing?
![image](https://user-images.githubusercontent.com/8092998/202068319-35b8e703-ea3b-4d9a-82b6-f736a881b833.png)

**Reply by philkim72** [reply]
As long as the most recent build under actions works, I don't think we need to worry about it.

**Reply by jtr13** [reply]
Correct, you can ignore failed older builds if the most recent one is working. Also, setup-r is not affected by any of our files, sometimes things just go wrong and the solution is to rerun.

Suggestion: it's better to post a link to the failed run rather than a screenshot so it's easier to navigate there.

**Comment by philkim72** [comment]
I have added the part headers for tutorials, cheatsheets and workshops.

**Comment by lauradls** [comment]
Hi Professor @jtr13 - Are medium blogposts allowed? Note that these are usually behind a paywall. https://medium.com/@yaohong010/python-tutorial-gather-data-through-api-and-web-scraping-f55de6000cac

**Reply by jtr13** [reply]
Yes, don't love the paywall but it's a common place for blogs and you can read a bunch free before the paywall kicks in.

**Reply by lauradls** [reply]
Thank you!  As a follow up, a student included the link to their cheatsheet in the ` .Rmd`, which is a pdf hosted on Github. When I go to that repository, there is only a pdf and not the code supporting it. Is this ok too? 
https://github.com/yuehanhu/ggplot2cheatsheet/blob/main/Cheatsheet.pdf

**Reply by jtr13** [reply]
Good question. Yes, that's ok. Cheatsheets are often created with Keynote or PowerPoint so there's no code.

**Reply by lauradls** [reply]
Thanks!

**Comment by rohansheelvant** [comment]
Hi everyone, 

The building of bookdown has failed due to "System command sh" failure. Does anyone have an idea why might this be happening?

PS: here is the link to failed run https://github.com/jtr13/cc22mw/actions/runs/3476326688/jobs/5811453058

**Comment by jtr13** [comment]
It looks like it had trouble finding an Ubuntu file, in short that has nothing to do with our code. Try clicking the re-run jobs button.

**Reply by rohansheelvant** [reply]
I did re-run the job. It fails due to a different error now. Here is the link: https://github.com/jtr13/cc22mw/actions/runs/3476326688/jobs/5811628796

**Comment by jtr13** [comment]
It's working now... go figure. Strange things happen with the install R and install R dependencies jobs that are beyond our control, best to just wait out the problems.

**Comment by jtr13** [comment]
Going forward it's best to open a new issue for each problem, keep discussion there, and then close it when the problem is resolved.

**Comment by lauradls** [comment]
Hi All! Does anyone know what we do with python packages? 
```import numpy as np```
```import plotly.graph_objects as go```

Should they go on directly on the import, on suggest, or is it not necessary?
